Flutter v2 documentation update (#473)
rudderlabs/rudderstack-docs: stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/index.mdx

<img src="../../../assets/event-stream-sources/flutter-sdk-1.png" alt="Flutter source write key" />

- You will also need a data plane URL. 

<div class="infoBlock">
  
For more information on the data plane URL and where to get it, refer to this <a href="https://www.rudderstack.com/docs/get-started/dashboard-overview/#data-plane-url" target="_blank">RudderStack Dashboard Overview</a> section.
</div>

- It is also recommended to set up the [**Flutter development environment**](https://flutter.dev/docs/get-started/install) in your system.

<div class="successBlock">

<strong>The RudderStack Flutter SDK is migrated to Null Safety starting from version 1.0.2.</strong>
</div>

## Installing the RudderStack Flutter SDK

The recommended way to install the Flutter SDK is through [`pub`](https://pub.dev/packages/rudder_sdk_flutter).

Follow the steps below to add the SDK as a dependency:

1. Open `pubspec.yaml` and add `rudder_sdk_flutter` under `dependencies` section, as shown:

```yaml
dependencies:
  rudder_sdk_flutter: ^2.0.1
```

2. Navigate to your application's root folder and install all the required dependencies with the following command:

```bash
flutter pub get
```

> ### Installing the SDK for the web

Author: Yep, as discussed these are additional steps to use the SDK for web. Clarified accordingly.

childProperty.put("test_child_key_1", "test_child_value_1");
property.put("test_key_2",childProperty);
rudderClient.track("test_track_event", properties: property);
```

The `track` method has the following signature:

| Name         | Data Type        | Required | Description                                                              |
| :----------- | :--------------- | :------- | :----------------------------------------------------------------------- |
| `name`       | String        | Yes      | Contains the name of the event you want to track.                         |
| `properties` | `RudderProperty` | No       | Contains the extra data properties you want to send along with the event. |
| `options`    | `RudderOption`   | No       | Contains the extra event options.                                        |

<div class="infoBlock">

RudderStack automatically tracks the following optional events:
<ol>
<li><code class="inline-code">Application Installed</code></li>
<li><code class="inline-code">Application Updated</code></li>
<li><code class="inline-code">Application Opened</code></li>
<li><code class="inline-code">Application Backgrounded</code></li>
</ol>

> You can disable these events by calling <code class="inline-code">withTrackLifeCycleEvents(false)</code> on the <code class="inline-code">RudderConfigBuilder</code> object while initializing the RudderStack client. However, we highly recommend keeping them enabled.

Reviewer: ```suggestion
You can disable these events by calling <code class="inline-code">withTrackLifeCycleEvents(false)</code> on the <code class="inline-code">RudderConfigBuilder</code> object while initializing the RudderStack client. However, it is highly recommended to keep them enabled.
```

rudderClient.reset();
```

> ## Enabling/disabling user tracking via the `optOut` API(GDPR support)

Reviewer: ```suggestion
## Enabling/disabling the tracking of user activities (GDPR support)
```

<div class="infoBlock">

This functionality is not available for the web. For more information, refer to the <a href="#flutter-sdk-api-not-available-for-web">Flutter SDK API not available for the web</a> section below.
</div>

> RudderStack gives the users \(e.g., an EU user\) the ability to opt out of tracking any user activity until the user gives their consent. You can do this by leveraging RudderStack's `optOut` API.

Reviewer: ```suggestion
RudderStack gives the users \(for example, an EU user\) the ability to opt out of tracking any user activity until the user gives their consent. You can do this by leveraging RudderStack's `optOut` API.
```
